src/wpt/harness/harness.ts

```diff
@@ -881,6 +889,76 @@ globalThis.assert_throws_dom = (
   );
 };
 
+/**
+ * Assert that a Promise is rejected with the right DOMException.
+ *
+ * For the remaining arguments, there are two ways of calling
+ * promise_rejects_dom:
+ *
+ * 1) If the DOMException is expected to come from the current global, the
+ * third argument should be the promise expected to reject, and a fourth,
+ * optional, argument is the assertion description.
+ *
+ * 2) If the DOMException is expected to come from some other global, the
+ * third argument should be the DOMException constructor from that global,
+ * the fourth argument the promise expected to reject, and the fifth,
+ * optional, argument the assertion description.
+ *
+ * @param test - the `Test` to use for the assertion.
+ * @param type - See documentation for
+ * `assert_throws_dom <#assert_throws_dom>`_.
+ * @param promiseOrConstructor - Either the constructor
+ * for the expected exception (if the exception comes from another
+ * global), or the promise that's expected to reject (if the
+ * exception comes from the current global).
+ * @param descriptionOrPromise - Either the
+ * promise that's expected to reject (if the exception comes from
+ * another global), or the optional description of the condition
+ * being tested (if the exception comes from the current global).
+ * @param [description] - Description of the condition
```

Comment:
Can you also make sure jsdoc param names match with the actual function param names?

```suggestion
 * @param [maybeDescription] - Description of the condition
```